shared/shared/postgresql_backend/pg_connection_handler.py

```diff
@@ -31,9 +32,12 @@ def wrapper(*args, **kwargs):
                     and e.base_exception.pgcode is None
                 ):
                     tries += 1
-                    if tries < MAX_RETRIES:
-                        continue
-                raise
+                    if tries >= MAX_RETRIES:
```

Comment:
Can you log each retry (maybe also with the reason -> the exception), so we can see how often it occurs in production?

Reply:
done